app/goal_routes.py

```diff
@@ -0,0 +1,160 @@
+from app import db
+from app import helper
+from .models.task import Task
+from .models.goal import Goal
+from flask import request, Blueprint, make_response, jsonify, Response
+from sqlalchemy import desc, asc
+from datetime import date
+import os
+
+#WAVE 5 - CRUD goal 
+#create new blueprint for goal
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+#WAVE 6 -
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"], strict_slashes=False)
+def create_goal_tasks(goal_id):
+    # get the goal for the specified goal_id
+    if not helper.is_int(goal_id):
+        return {
+            "message": "id must be an integer",
+            "success": False
+        },400
+    
+    goal = Goal.query.get(goal_id)
+    
+    if (goal is None):
+        return jsonify(details=f"No Goal found with ID: '{goal_id}'"), 400     
+    
+    # get the tasks from the request body
+    form_data = request.get_json()
+
+    taskids = form_data["task_ids"]
+
+    # get the the tasks with the specified ids
+    for tid in taskids:
+        task = Task.query.get(int(tid))
+        
+        if (task is not None):
+            task.goal_id = goal_id
+            db.session.add(task)    
+        else:
+            # this handles scenarios where specified TaskId in the request body in invalid / does not exist
+            return jsonify(details=f"No Task found with ID: '{tid}'"), 400 
+    
+    db.session.commit()
+    
+    return jsonify(id=int(goal_id), task_ids=taskids), 200 #int removed the quotes from the goal_id
```

Comment:
This change would have made the above-mentioned bug more obvious. This would also allow the response to include ALL tasks associated with the goal, not just the new ones.
This isn't super intuitive so I'd definitely be down to talk through this more in our next 1:1

```suggestion
    full_task_ids = []
    for task in goal.tasks:
        full_task_ids.append(task.task_id)
        
    return jsonify(id=int(goal_id), task_ids=full_task_ids), 200 #int removed the quotes from the goal_id
```